tests/libstat_wrapper.c

```diff
@@ -0,0 +1,35 @@
+/*
+ * Simulate FORCESOCKETSFILE existing for the IPC tests
+ */
+
+#include <stdio.h>
+#include <dlfcn.h>
+#include <string.h>
+#include <sys/stat.h>
+#include "../include/config.h"
+#if defined(QB_LINUX) || defined(QB_CYGWIN)
+#include <gnu/lib-names.h>
+#endif
+
+int __xstat(int __ver, const char *__filename, struct stat *__stat_buf)
+{
+#if defined(QB_LINUX) || defined(QB_CYGWIN)
+	static int opened = 0;
+	static void *dlhandle;
+	static int (*real_xstat)(int __ver, const char *__filename, void *__stat_buf);
+
+	if (!opened) {
+		dlhandle = dlopen(LIBC_SO, RTLD_NOW);
```

Comment:
Had been doing that in the past with dlopen_fatal(RTLD_NEXT, ...) at least in the assumption it would then get me the symbol coming next in the search order instead of directly jumping to libc.
And instead of keeping the handle in a static variable (when using an explicit library - with RTLD_NEXT there shouldn't be a need) I closed it after getting the symbols I needed.

Reply:
Sry for the above - that is of course dlsym_fatal.
And that again is just my wrapper for dlsym exiting out if it doesn't find the symbol.
The comment in parentheses already points to something being fishy here ;-)